File: backend/src/services/OperationLogService.cpp
#include "services/OperationLogService.h"
#include "repositories/MockOperationLogRepository.h"
#include "utils/TimeUtil.h"

namespace smart_home {

OperationLog OperationLogService::write(long long userId, long long homeId, const std::string& actionType,
                                        const std::string& actionDesc, const std::string& targetType,
                                        long long targetId, const std::string& ipAddress) {
    MockOperationLogRepository repo;
    OperationLog log;
    log.userId = userId;
    log.homeId = homeId;
    log.actionType = actionType;
    log.actionDesc = actionDesc;
    log.targetType = targetType;
    log.targetId = targetId;
    log.ipAddress = ipAddress;
    log.createdAt = TimeUtil::now();
    return repo.save(log);
}

std::vector<OperationLog> OperationLogService::listByHome(long long homeId) {
    MockOperationLogRepository repo;
    return repo.findByHome(homeId);
}

} // namespace smart_home

File: backend/app/services/operation_log_service.py
from app.repositories.mock.operation_log_mock import MockOperationLogRepository


class OperationLogService:
    def __init__(self):
        self.logs = MockOperationLogRepository()

    def write(self, user_id: int, home_id: int | None, action: str, description: str, target_type: str | None = None, target_id: int | None = None) -> dict:
        return self.logs.write(user_id, home_id, action, description, target_type, target_id).__dict__

    def list_by_home(self, home_id: int) -> list[dict]:
        return [log.__dict__ for log in self.logs.list_by_home(home_id)]